example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,370 @@
+
+    if (child != null) {
+      context.paintChild(child!, offset + contentOffset);
+    }
+  }
+
+  @override
+  bool hitTest(BoxHitTestResult result, {required Offset position}) {
+    if (hitTestChildren(result, position: position)) {
+      return true;
+    }
+    // Allow hit-testing around the content, e.g, we might have padding and
+    // the user is trying to drag using the padding area.
+    final rect = Offset.zero & size;
+    return rect.contains(position);
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    final direction = _computeArrowDirection(Offset.zero & size, focalPoint);
+    final contentOffset = _computeContentOffset(direction, arrowLength);
+
+    return result.addWithPaintOffset(
+      offset: contentOffset,
+      position: position,
+      hitTest: (BoxHitTestResult result, Offset transformed) {
+        assert(transformed == position - contentOffset);
+        return child?.hitTest(result, position: transformed) ?? false;
+      },
+    );
+  }
+
+  /// Builds the path used to paint the menu.
+  ///
+  /// The path includes a rounded rectangle and a arrow pointing to [arrowDirection], centered at [arrowCenter].
+  Path _buildBorderPath(ArrowDirection arrowDirection, double arrowCenter) {
+    final halfOfBase = arrowBaseWidth / 2;
+
+    // Adjust the rect to leave space for the arrow.
+    // During layout, we reserve space for the arrow in both x and y axis.
+    final contentRect = Rect.fromLTWH(
+      arrowLength,
+      arrowLength,
+      size.width - arrowLength * 2,
+      size.height - arrowLength * 2,
+    );
+
+    Path path = Path()..addRRect(RRect.fromRectAndRadius(contentRect, Radius.circular(borderRadius)));
+
+    // Add the arrow.
+    if (arrowDirection == ArrowDirection.left) {
+      path
+        ..moveTo(contentRect.centerLeft.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.right) {
+      path
+        ..moveTo(contentRect.centerRight.dx, arrowCenter - halfOfBase)
+        ..relativeLineTo(arrowLength, halfOfBase)
+        ..relativeLineTo(-arrowLength, halfOfBase);
+    } else if (arrowDirection == ArrowDirection.up) {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.topCenter.dy)
+        ..relativeLineTo(halfOfBase, -arrowLength)
+        ..relativeLineTo(halfOfBase, arrowLength);
+    } else {
+      path
+        ..moveTo(arrowCenter - halfOfBase, contentRect.bottomCenter.dy)
+        ..relativeLineTo(halfOfBase, arrowLength)
+        ..relativeLineTo(halfOfBase, -arrowLength);
+    }
+
+    path.close();
+
+    return path;
+  }
+
+  /// Computes the direction where the arrow should point to.
+  ///
+  /// If [globalFocalPoint] is inside the [menuRect] horizontal bounds, the arrow points up or right. Otherwise,
```

Comment:
I don't understand this rule. The `menuRect` is the rectangle that bounds the menu content, right? If so, why would the focal point be inside of it? And why would that be the only circumstance in which the arrow points up/down?

Reply:
For example:

```
      [x]
 --------------
   menu here
---------------
```

```
[x]
        ----------------
           menu here
        ----------------
```

In the first example the arrow will point up, in the second one the arrow will point left.

That's what I meant by "inside menuRect horizontal bounds"